pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,408 @@
+package org.broadinstitute.dsm.db;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.structure.ColumnName;
+import org.broadinstitute.dsm.db.structure.TableName;
+import org.broadinstitute.dsm.model.elastic.export.Exportable;
+import org.broadinstitute.dsm.model.elastic.export.painless.PutToNestedScriptBuilder;
+import org.broadinstitute.dsm.model.elastic.export.painless.UpsertPainlessFacade;
+import org.broadinstitute.dsm.model.filter.postfilter.HasDdpInstanceId;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+@Data
+@TableName(name = DBConstants.SOMATIC_DOCUMENTS_TABLE,
+        alias = DBConstants.SOMATIC_DOCUMENTS_TABLE_ALIAS,
+        primaryKey = DBConstants.SOMATIC_DOCUMENTS_PK,
+        columnPrefix = "")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SomaticResultUpload implements HasDdpInstanceId {
+    @ColumnName(ESObjectConstants.SOMATIC_DOCUMENT_ID)
+    private final Long somaticDocumentId;
+
+    @ColumnName(DBConstants.DDP_INSTANCE_ID)
+    private final Long ddpInstanceId;
+
+    private final String ddpParticipantId;
+
+    @ColumnName(DBConstants.PARTICIPANT_ID)
+    private final Long participantId;
+
+    @ColumnName(DBConstants.FILE_NAME)
+    private final String fileName;
+
+    @ColumnName(DBConstants.MIME_TYPE)
+    private final String mimeType;
```

Comment:
Why is mimeType needed? Or is it for future reference in case we need to know the type of the file?

Reply:
It's a future placeholder incase the file type changes.

Reply:
What would we do differently based on the file type?

Reply:
FWIW, I'm using the value supplied for validation so it is memorializing information provided when the file was uploaded but otherwise doesn't really matter.  It's easy to remove.

Reply:
My understanding from DM is that we are using mimetype to verify file type. Although we only have one type now, this would be useful in a future where we store other types of somatic result files. 
Your call on the likelihood of that future.